### ISSUE TITLE
Krita export layers to json file

### ISSUE THREAD

**Issue by songproducer** [ndee85/coa_tools#42]

**Comment by ndee85** [comment]
Once Krita has a scripting API I'll add an exporter for it. Last kickstarter campaign funded scripting as one new goal. So this is just a matter of time.

**Comment by blurymind** [comment]
Thank you @ndee85 
I am looking forward to Krita's scripting API as well

**Comment by manuelsongokuh** [comment]
hello, sorry i dont understand..

so i want to know how i will export from krita to json for blender?

**Comment by OBKF** [comment]
@manuelsongokuh For now just export .ora from Krita to gimp and then use Gimp exporter to export that json to blender.

**Comment by manuelsongokuh** [comment]
yes i see this i did done it but not from krita to export.. maybe in the future..so thank you

**Comment by Larpon** [comment]
https://krita.org/en/item/look-new-presets-another-krita-4-development-build/ <- Python scripting finally available in the Linux `appimage` builds!

**Comment by ndee85** [comment]
Yes.. I am planning to do this.
Making myself familiar with the krita scripting API.

**Comment by Larpon** [comment]
I made a little donation to get you started :)

**Comment by ndee85** [comment]
I've seen it. Thanks a bunch for the support :-) :+1:

**Comment by manuelsongokuh** [comment]
thank you @ndee85 

you can open crowdfund or Kickstarter for you developing this coatool..?

**Comment by ndee85** [comment]
I actually have a patreon page. But I want to make it more prominent in the future. Planning on creating a dedicated website for coa tools. And Patreon will be more visible on that page. :-)

**Comment by manuelsongokuh** [comment]
@ndee85 where i find your patreon..? link?

**Comment by ndee85** [comment]
https://www.patreon.com/andreasesau

**Comment by manuelsongokuh** [comment]
i did start donation to you patreon. and i nave send message to you.. thank you

**Comment by Fireruner** [comment]
This would be amazing if you could do it! :D Thanks

**Comment by vini-guerrero** [comment]
Can this be used within COA Tools?

https://github.com/GDquest/GDquest-art-tools

**Comment by NathanLovato** [comment]
@vini-guerrero not now, but someone could contribute the code to export the JSON COA uses. We already have the code to generate the folder and file structure, trim the images...

**Comment by Larpon** [comment]
Hmm. I made a JSON exporter a while back in Krita - but I ended up struggling with the UI for it - I'll try to and see what I can contribute

**Comment by Larpon** [comment]
I've made a more general solution to export document and layer meta data in different formats.
But it's a mess right now - and should be cleaned up. I've hacked a layer export into as well but stopped because Krita lacked some scripting functionality I needed
![image](https://user-images.githubusercontent.com/768942/53695993-bcae4c80-3dc2-11e9-8537-bb740131ae5a.png)

**Comment by vini-guerrero** [comment]
I have a friend who was able to run COA_Tools in Godot 3.1 with a few changes in source in the code, however the JSON file he used from Photoshop, would be nice if we could match the same json with Krita.
That one seems really good @Larpon nice job!

**Comment by NathanLovato** [comment]
@Larpon Nice! Would you consider contributing some code to our FOSS add-on? https://github.com/GDquest/GDquest-art-tools

We have a flexible batch export system in place, wrapper classes to help work with Krita's layers much more easily than by default. The ability to export wanted layers as JSON would be great, especially if it means adding support for COA tools or tools like e.g. Spine.

Razvan worked around the limitations of Krita's export system, and he's also fixed some issues in Krita's Python API for Krita 4.2

**Comment by Larpon** [comment]
@NathanLovato yeah sure - I'll check your stuff out today 🙂

**Comment by Larpon** [comment]
Just leaving a note here to say that I  contributed rudimentary COA Tools export to GDquest-art-tools. I'll be happy to contribute to this project as well. I think it should be done without patching Krita though - which I think might require some dialogue with the krita dev team.

**Comment by faridosc** [comment]
Is there any workaround to make this work?

**Comment by faridosc** [comment]
This is working and can be closed.

**Comment by Larpon** [comment]
Yes. There's also ongoing efforts to include and ship the exporter python dependencies with Krita!

**Comment by Larpon** [comment]
**Update** The exporter is now in [Krita master branch](https://invent.kde.org/graphics/krita/-/tree/master/plugins/python/batch_exporter)

**Comment by j2l** [comment]
It's borken for Krita 5.0.2 and 5.2.2 (Ubuntu 22.04)
